### PR TITLE
Update Discrete Noise Distribution

### PR DESCRIPTION
**Pull request by JohannesHuegle**
- Define new noise distribution with appropriate default
- Provides faithful distrbutions under H_0 and H_1

### CODE REVIEW

File: manm_cs/prob_distributions/discrete/uniform_discrete_distribution.py [hunk 1]
```diff
@@ -10,4 +10,4 @@ def __init__(self, num_values: int):
         super(UniformDiscreteDistribution, self).__init__(num_values=num_values)
 
     def sample(self, num_observations: int) -> np.array:
-        return np.random.randint(low=0, high=self.num_values - 1, size=num_observations)
+        return np.random.randint(low=0, high=self.num_values, size=num_observations)
```
Comment (ChristopherSchmidt89) [inline manm_cs/prob_distributions/discrete/uniform_discrete_distribution.py:13]: good spotting, that was indeed incorrect, as num values = 2 would have caused a discrete domain of 1

File: manm_cs/noise/discrete_noise_builder.py [hunk 2]
```diff
@@ -15,9 +15,14 @@ def with_num_discrete_values(self, num_discrete_values: int) -> 'DiscreteNoiseBu
         return self
 
     def build(self) -> DiscreteNoise:
+        # Portion transmitted to uniform noise, i.e., signal=0 for no noise and signal =1 for uniform noise
         signal = self.signal_to_noise_ratio
+        
+        # Portions of unfiform noise over all discrete values
```
Comment (ChristopherSchmidt89) [inline manm_cs/noise/discrete_noise_builder.py:21]: correct spelling of unfiform

File: manm_cs/noise/discrete_noise_builder.py [hunk 3]
```diff
@@ -15,9 +15,14 @@ def with_num_discrete_values(self, num_discrete_values: int) -> 'DiscreteNoiseBu
...
+        signal_noise_classes = signal / self.num_discrete_values
```
Comment (ChristopherSchmidt89) [inline manm_cs/noise/discrete_noise_builder.py:22]: is it necessary to introduce `signal`, or could you directly work with `self.signal_to_noise_ratio` instead

File: manm_cs/noise/discrete_noise_builder.py [hunk 4]
```diff
@@ -15,9 +15,14 @@ def with_num_discrete_values(self, num_discrete_values: int) -> 'DiscreteNoiseBu
...
         num_noise_classes = self.num_discrete_values - 1
-        avg_noise = (1 - signal) / num_noise_classes
-        probs = [signal] + [avg_noise for _ in range(num_noise_classes)]
+        
+        # Probabilities
+        probs = [(1-signal)+signal_noise_classes] + [(signal_noise_classes) for _ in range(num_noise_classes)]
```
Comment (ChristopherSchmidt89) [inline manm_cs/noise/discrete_noise_builder.py:26]: same here `signal` vs `self.signal_to_noise_ratio`

Comment (ChristopherSchmidt89) [reply manm_cs/noise/discrete_noise_builder.py:26]: similar comment for the `num_noise_classes` vs `self.num_discrete_values - 1`

**Review by ChristopherSchmidt89 — CHANGES_REQUESTED** [review]
Mostly fine. However we should discuss the use of `max_discrete_value_classes` as I am not sure if we have the same understanding, or it is correctly defined as an input

File: manm_cs/graph/graph_builder.py [hunk 5]
```diff
@@ -162,7 +162,7 @@ def sample_beta(self) -> List[float]:
     def generate_discrete_variable(self, parents, node_idx) -> 'DiscreteVariable':
         num_values = np.random.randint(
             low=self.min_discrete_value_classes,
-            high=self.max_discrete_value_classes + 1,
+            high=self.max_discrete_value_classes,
```
[the remark this answers is not in the document]
Comment (JohannesHuegle) [reply manm_cs/graph/graph_builder.py:165]: correct - thanks

**Review by ChristopherSchmidt89 — APPROVED** [review]
thanks for changes

**Pull request merged**